wordpress-mobile/WordPress-iOS
# Prepare suggestions to handle xposts

Addresses https://github.com/wordpress-mobile/gutenberg-mobile/issues/2602

Prepares `SuggestionsTableView` and friends — which are currently made to only work with @-mentions — to handle other types of suggestions we're adding in the very near future, such as xposts.

- Update `SuggestionsTableView.m/.h` to use a more succinct initializer
- Add new logic to `SuggestionsTableView.swift` to handle multiple mention types (i.e. mentions, and in the future, xposts)
- Rename variable names in `SuggestionsTableViewCell.m/.h` to be @-mention agnostic
- Rename `GutenbergMentionsViewController` to `GutenbergSuggestionsViewController` to reflect its role supporting not just mentions, but also xposts in the near future
- Rename variable names in `GutenbergViewController` to be @-mention agnostic

### To test

Smoke test @-mentions in each of the following:

```
- [ ] Reader comments
- [ ] Comment on post (via notification)
- [ ] Reply to comment (via notification)
- [ ] Gutenberg editor mentions
```

<details>
<summary>Test details</summary>
<br>

**What you'll need**: A WordPress.com site that you can comment on and that has one or more other WordPress.com users (in order to test @-mentions). Make sure some of the users have a profile photo. The user that you login with will also need:
- a notification about a post
- a notification about a comment
<br>

**Reader comments**

<ol>
<li>In the Reader tab, locate the site</li>
<li>Navigate to a post, tap the comment button</li>
<li>Tap the &quot;Reply to post...&quot; text box to bring up the comment UI</li>
<li>Type <code>@</code> and verify that the list of suggestions is shown</li>
<li>Make sure that the avatar images load for users who have a profile photo</li>
<li>Select one of the suggestions to add it to your comment and hit send</li>
<li>Verify that the mentioned user receives a notification</li>
</ol>

**Comment on post (via notification)**

<ol>
<li>In the Notifications tab, locate a notification about a post</li>
<li>Open it, then tap the comment button, then the &quot;Reply to post...&quot; text box to bring up the comment UI</li>
<li>Type <code>@</code> and verify that the list of suggestions is shown</li>
<li>Make sure that the avatar images load for users who have a profile photo</li>
<li>Select one of the suggestions to add it to your comment and hit send</li>
<li>Verify that the mentioned user receives a notification</li>
</ol>

**Reply to comment (via notification)**

<ol>
<li>In the Notifications tab, locate a notification about a comment</li>
<li>Open it, then tap the &quot;Write a reply...&quot; text box to bring up the comment UI</li>
<li>Type <code>@</code> and verify that the list of suggestions is shown</li>
<li>Make sure that the avatar images load for users who have a profile photo</li>
<li>Select one of the suggestions to add it to your reply and hit send</li>
<li>Verify that mentioned user receives a notification</li>
</ol>

**Gutenberg editor mentions**

<ol>
<li>Open the block editor</li>
<li>In the &quot;Start writing...&quot; section, type <code>@</code> and verify that the list of suggestions is shown</li>
<li>Make sure that the avatar images load for users who have a profile photo</li>
<li>Select one of the suggestions and publish the post</li>
<li>Verify that mentioned user receives a notification</li>
</ol>
</details>


### PR submission checklist
